### PR TITLE
PCM-3710: Moving CSV Export to SOLR

### PR DESCRIPTION
**Pull request by jtrantin**
@gandhikeyur @vrathee @engineersamuel Please review.

https://projects.engineering.redhat.com/browse/PCM-3710

Moving the CSV export to SOLR speeds it up a lot, thanks to strata mapping layer the output is identical as before.

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
-  "presets": ["es2015"]
-//  "plugins": [
-//    "add-module-exports"
-//  ]
+    "presets": ["es2015"],
+    "plugins": [
+//        "add-module-exports",
+        "transform-async-to-generator"
```
Comment (jtrantin) [inline .babelrc]: Added this so we can use async/await in PCM.

File: app/cases/views/list.jade [hunk 2]
```diff
@@ -22,7 +22,7 @@ div(ng-hide='showCaseList()')
         span(ng-show="exporting")
           span.spinner.spinner-inline.spinner-sm
           span(translate) Exporting CSV
-        a.href(ng-click='exports()', ng-hide='ie8 || ie9 || exporting || RHAUtils.isNotEmpty(CaseService.bookmarkedAccount)', translate='') Export as CSV
+        a.href(ng-click='exports()', ng-hide='ie8 || ie9 || exporting || RHAUtils.isNotEmpty(CaseService.bookmarkedAccount)', translate='') Export all account cases as CSV
```
Comment (jtrantin) [inline app/cases/views/list.jade]: Not 100% sure on the wording here. It needs to be clear users export all cases from their account, not just the ones visible on the list below. It also needs to be short, so something like "Export all cases from your account as CSV" is not applicable.

File: package.json [hunk 3]
```diff
@@ -1,108 +1,110 @@
...
+    "angular-route": "~1.5.7",
+    "angular-sanitize": "~1.5.7",
+    "angular-scenario": "~1.5.7",
+    "angular-ui-bootstrap": "~1.3.3",
+    "angular-ui-router": "~0.3.1",
+    "autoprefixer": "~6.3.7",
+    "babel-core": "~6.10.4",
+    "babel-loader": "~6.2.4",
+    "babel-plugin-transform-async-to-generator": "^6.22.0",
+    "babel-polyfill": "^6.22.0",
```
Comment (jtrantin) [inline package.json]: Just added `babel-plugin-transform-async-to-generator` and `babel-polyfill`.

**Comment by renujhamtani** [comment]
Looks good

File: app/cases/views/list.jade (hunk 2 above)
Comment (vrathee) [inline app/cases/views/list.jade]: Maybe we can give details in tooltip and link can just say "Export as CSV"

**Comment by vrathee** [comment]
Apart from showing tool-tip.. everything looks Good.. Reviewed.

File: app/cases/views/list.jade (hunk 2 above)
Comment (jtrantin) [inline app/cases/views/list.jade]: Good idea, I'll change that.

Comment (jtrantin) [inline app/cases/views/list.jade]: Done, now it looks like https://goo.gl/photos/cW24NZaLZ1n6Vvxs8

File: app/cases/controllers/list.js [hunk 4]
```diff
@@ -19,29 +21,18 @@ export default class List {
...
+        $scope.exports = async function () {
+            $scope.exporting = true;
 
+            try {
+                const query = `case_accountNumber:${this.securityService.loginStatus.authedUser.account.number}`;
+                const response = await strataService.cases.advancedSearch(query, null, 0, 10000, 'csv');
+                const csvBlob = new Blob([response], {type: 'text/csv'});
+                FileSaver.saveAs(csvBlob, 'caseList.csv');
+            } catch (e) {
+            }
```
Comment (engineersamuel) [inline app/cases/controllers/list.js]: No Alert if there is an error here?

Comment (jtrantin) [inline app/cases/controllers/list.js]: You are right, not sure why I though none is necessary, even the old code shows a msg.

Comment (jtrantin) [inline app/cases/controllers/list.js]: Fixed

**Comment by jtrantin** [comment]
Fixed missing alert and added tooltip to the link.

**Pull request merged**